mayastor/src/lvs/lvol.rs

```diff
@@ -259,6 +275,13 @@ impl Lvol {
         let blob = unsafe { self.0.as_ref().blob };
         assert_ne!(blob.is_null(), true);
 
+        if self.is_snapshot() {
```

Comment:
It's good we check errors here but perhaps we should be more strict and assert? in lvs_pool.rs we have code something like:

```rust
pub fn lvols(&self) -> Option<impl Iterator<Item = Lvol>> 
``` 

there we filter for out the lvols we belonging to a pool, what if we add a new add a filter there? i.e where we check if its a snapshot or not? then we can assert here and leave out the warning as it seems to me that never want to call `set()`  on a snapshot?

Reply:
This filter is used in [un]share_all so it won't help when only a single replica is being shared via `pool_grpc::share_replica`, unless I'm missing something.